DuckDuckGo/RemoteMessagingConfigMatcherProvider.swift

```diff
@@ -0,0 +1,131 @@
+
+        var bookmarksCount = 0
+        var favoritesCount = 0
+        let context = bookmarksDatabase.makeContext(concurrencyType: .privateQueueConcurrencyType)
+        context.performAndWait {
+            bookmarksCount = BookmarkUtils.numberOfBookmarks(in: context)
+            favoritesCount = BookmarkUtils.numberOfFavorites(for: appSettings.favoritesDisplayMode, in: context)
+        }
+
+        let statisticsStore = StatisticsUserDefaults()
+        let variantManager = DefaultVariantManager()
+        let subscriptionManager = AppDependencyProvider.shared.subscriptionManager
+
+        let isPrivacyProSubscriber = subscriptionManager.accountManager.isUserAuthenticated
+        let isPrivacyProEligibleUser = subscriptionManager.canPurchase
+
+        let activationDateStore = DefaultVPNActivationDateStore()
+        let daysSinceNetworkProtectionEnabled = activationDateStore.daysSinceActivation() ?? -1
+
+        var privacyProDaysSinceSubscribed: Int = -1
+        var privacyProDaysUntilExpiry: Int = -1
+        var privacyProPurchasePlatform: String?
+        var isPrivacyProSubscriptionActive: Bool = false
+        var isPrivacyProSubscriptionExpiring: Bool = false
+        var isPrivacyProSubscriptionExpired: Bool = false
+        let surveyActionMapper: DefaultRemoteMessagingSurveyURLBuilder
+
+        if let accessToken = subscriptionManager.accountManager.accessToken {
+            let subscriptionResult = await subscriptionManager.subscriptionEndpointService.getSubscription(
+                accessToken: accessToken
+            )
+
+            if case let .success(subscription) = subscriptionResult {
+                privacyProDaysSinceSubscribed = Calendar.current.numberOfDaysBetween(subscription.startedAt, and: Date()) ?? -1
+                privacyProDaysUntilExpiry = Calendar.current.numberOfDaysBetween(Date(), and: subscription.expiresOrRenewsAt) ?? -1
+                privacyProPurchasePlatform = subscription.platform.rawValue
+
+                switch subscription.status {
+                case .autoRenewable, .gracePeriod:
+                    isPrivacyProSubscriptionActive = true
+                case .notAutoRenewable:
+                    isPrivacyProSubscriptionExpiring = true
+                case .expired, .inactive:
+                    isPrivacyProSubscriptionExpired = true
+                case .unknown:
+                    break // Not supported in RMF
+                }
+
+                surveyActionMapper = DefaultRemoteMessagingSurveyURLBuilder(statisticsStore: statisticsStore, subscription: subscription)
+            } else {
+                surveyActionMapper = DefaultRemoteMessagingSurveyURLBuilder(statisticsStore: statisticsStore, subscription: nil)
+            }
+        } else {
+            surveyActionMapper = DefaultRemoteMessagingSurveyURLBuilder(statisticsStore: statisticsStore, subscription: nil)
+        }
+
+        let dismissedMessageIds = store.fetchDismissedRemoteMessageIds()
```

Comment:
tiny thing, but I think we once established these acronyms should be all capitalized, so IDs

Reply:
yeah, I personally hate that too :D just haven't updated it because it's been like that. Happy do to it since I've moved/updated the entire protocol anyway. Thanks for pointing it out!